packages/flutter/lib/src/material/icon_button.dart

```diff
@@ -38,6 +38,8 @@ class IconButton extends StatelessWidget {
   const IconButton({
     Key key,
     this.size: 24.0,
+    this.padding: const EdgeInsets.all(8.0),
+    this.alignment: FractionalOffset.centerLeft,
```

Comment:
Just a convenience (OK a hack), since in the only AppBar case where the alignment matters it needs to be centerLeft. You're right that the default should just be center. I'll fix this.